components/shared_code/src/shared_data_model/meta/source.py

```diff
@@ -1,103 +1,44 @@
 """Data model source model."""
 
-from __future__ import annotations
+from typing import Self
 
-import pathlib
-from typing import cast
+from pydantic import Field, HttpUrl, SerializeAsAny, model_validator
 
-from pydantic import Field, HttpUrl, validator
-
-from .base import DescribedModel, MappedModel, NamedModel
-from .entity import Entities
-from .parameter import Parameter, Parameters
+from .base import DescribedModel, NamedModel
+from .entity import Entity
+from .parameter import Parameter
 
 
 class Configuration(NamedModel):
     """Configuration for specific metrics."""
 
-    metrics: list[str] = Field(..., min_items=1)
-    value: list[str] = Field(..., min_items=1)
-
-
-class Configurations(MappedModel[Configuration]):
-    """Source configurations."""
-
-
-class Documentation(MappedModel[str]):
-    """Source documentation for specific metrics."""
+    metrics: list[str] = Field(..., min_length=1)
+    value: list[str] = Field(..., min_length=1)
 
 
 class Source(DescribedModel):
     """The source model extends the base model with source parameters and measurement entities."""
 
     url: HttpUrl | None = None
-    documentation: Documentation | None = None  # Documentation in Markdown format
-    configuration: Configurations = cast(Configurations, {})
-    parameters: Parameters
-    entities: Entities = cast(Entities, {})
+    documentation: dict[str, str] | None = None  # Documentation in Markdown format
+    configuration: dict[str, Configuration] = {}
+    parameters: dict[str, SerializeAsAny[Parameter]]
```

Comment:
For `SerializeAsAny` see https://docs.pydantic.dev/latest/usage/serialization/#subclass-instances-for-fields-of-basemodel-dataclasses-typeddict.

Reply:
Do we want to further refactor this in the future, or leave it as duck typing?

Reply:
We would need to refactor the Parameter class hierarchy to something that uses composition I guess, but it isn't immediately obvious to me how that would work.